### ISSUE TITLE
Issue Exporting Shapes on Flash CS6 on MacOS 10.10.5

### ISSUE THREAD

**Issue by 23b** [TomByrne/Flash2Svg#18]
Hi There!
This is the first time i try to use Flash2Svg. 
I have sucessfully exported a SVG file with only Text on Stage and Text to Outline disabled.

Exporting any simpel Shape will cause the Export Progress to stall. (No CPU load though).
Here is the Error Message on the Output Panel:

```
TypeError: svgXml.defs[0] has no properties
SVG Export Failed (when i press cancel)
```

This Process leaves me with Temp_X Folders in my Flash Library containing Shapes and an empty file_temp Folder on the Filesystem instead of an exported SVG FIle.

Thanks a lot for this little tool, would be great to get this working!

**Comment by Wojciechem** [comment]
Hey, having the same issue in CS 5.5 / Windows 7, manual install. I'm trying to export single library item with no "dependencies", all settings default.

svgXml.defs[0] is first accessed [here](https://github.com/TomByrne/Flash2Svg/blob/cc/html/ExtensionContent/Extensible/SVG.jsfl#L656).
stupid "commenting out" lines 656-666 results in no error for me (but the exported file does not contain the shape, ie. is blank:)
`<?xml version="1.0" encoding="utf-8"?><svg xmlns:xlink="http://www.w3.org/1999/xlink" id="poziom1" image-rendering="auto" baseProfile="basic" version="1.1" x="0px" y="0px" width="1200" height="1000" xmlns="http://www.w3.org/2000/svg"><g id="elementy-sta.__C5.__82e_windaf0" overflow="visible"/></svg>`

I'm also looking forward to get it working :)

**Comment by TomByrne** [comment]
Would either of you be able to attach the offending FLA and I'll take a look.

**Comment by Wojciechem** [comment]
I created a new sample one, throwing the same error on export.
[example.txt](https://github.com/TomByrne/Flash2Svg/files/296301/example.txt)
(please rename the file extension, can't attach it as .fla because GitHub)

thank you for the reply.

**Comment by SeanJD** [comment]
Hi
I'm also having this issue - I hit Export and get 

> TypeError: svgXml.defs[0] has no properties

Then nothing happens. I wait and press cancel and get

> SVG Export Failed

in the output window.

This is on Windows 10, Flash CS6 and a new fla with a simple movieclip symbol on the timeline with a square in it (no animation) but the same thing happens no matter how simple/complex my timeline/animation.

**Comment by 23b** [comment]
Here is a hint: Releases up to v3.36 are working on my system. From v3.40 onwards i get the error.

**Comment by SeanJD** [comment]
I did think of trying the archived versions, but didn't! Thanks for the
heads up 23b I'll try that and see if it works

**Comment by SeanJD** [comment]
Yes it now works (with v3.36) but didn't with the latest.

**Comment by Wojciechem** [comment]
Slightly off-topic: You can export non-animated SVG-s from Flash if you happen to have Illustrator.
File -> Export -> As Image -> fxg. FXG can be opened by Illustrator and saved as SVG.
Hope I'm not being rude by posting this here :)

**Comment by TomByrne** [comment]
Hey,
I've looked into this.
At the beginning of the year I migrated from my own SVG path generator (which didn't cope well with certain shapes) to the built-in SVG generator in Flash CC (introduced in 2014 edition).
This improved speed and reliability, but it means that the tool no longer works in CC 2013 and earlier.
I wasn't aware that this export option was only added in 2014.

An option to add support back in would be to export to FXG in these versions and then convert the generated FXG to SVG. This would all happen behind the scenes, so the user would be totally unaware.
That said, it would probably take about 14hrs to do, so I doubt I'll find the time (without a financial backer).

In the meantime I guess CS5.5, CS6 & CC 2013 users have to use v3.36.
